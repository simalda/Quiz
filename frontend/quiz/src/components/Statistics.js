import React from "react";

import { connect } from "react-redux";
import * as ThunkActions from "../redux/thunkActions";

class ConnectedStatistics extends React.Component {
  constructor(props) {
    super(props);
    this.resultList = [];
  }
  correctAn(i) {
    let chosen;
    if (this.props.userStat[i]["ChosenAnswerId"] === 0) {
      chosen = (
        <div>Chosen answer: {this.props.userStat[i]["correctAnswer"]}</div>
      );
    } else {
      chosen = (
        <div>Chosen answer: {this.props.userStat[i]["ChosenAnswerId"]}</div>
      );
    }
    return chosen;
  }
  render() {
    let i = 0;
    while (i < this.props.userStat.length) {
      this.resultList.push(
        <div key={i} className="">
          <h5>Question: {i + 1}</h5>
          <div>{this.props.userStat[i]["Question"]}</div>
          <div>Correct answer: {this.props.userStat[i]["correctAnswer"]}</div>
          {this.correctAn(i)}
          <div>
            -------------------------------------------------------------------------------------
          </div>
        </div>
      );
      i++;
    }

    return (
      <div className="row">
        <div className="col-lg-3"></div>
        <div className="col-lg-6">
          <h3 id="quizNumber">QUIZ {this.props.userStat[0]["QuizId"]}</h3>
          {this.resultList}
        </div>
        <div className="col-lg-3"></div>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps,
    userStat: state.modalReducer.userStat
  };
};

const mapDispatchToProps = dispatch => {
  return {
    statistics: user => {
      dispatch(ThunkActions.statistics(user));
    }
  };
};

const Statistics = connect(
  mapStateToProps,
  mapDispatchToProps
)(ConnectedStatistics);

export default Statistics;
