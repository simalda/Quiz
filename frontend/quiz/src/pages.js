const pages = {
    Login: "Login",//-1
    NumberofQuestions:"NumberofQuestions",//3
    Main: "Main",//1
    QuizStep: "QuizStep",//0
    Results:"Results",//2
    Statistics: "Statistics",//5
    AddQuestion:"AddQuestion",
    AboutUs: "AboutUs"
  };
  
  export default pages