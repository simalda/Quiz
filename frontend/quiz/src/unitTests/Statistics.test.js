import React from "react";
import ReactDOM from "react-dom";

import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Statistics from "../components/Statistics";
import pages from "../JS/pages";

import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();
const store = mockStore({});
Enzyme.configure({ adapter: new Adapter() });

const state = {
  quiz: [
    {
      question: "a",
      answer: ["c", "d", "s"],
      correctAnswer: "cor"
    },
    {
      question: "ab",
      answer: ["c", "d", "s"],
      CorrectAnswer: "cor2"
    }
  ],
  highlight: "none",
  page: pages.Statistics,
  userStat: [{ QuizId: 1 }, { QuizId: 3 }],
  answerOptions: [],
  resultList: []
};

describe.skip("Statistics", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(
      <Provider store={store}>
        <Statistics submit={mockSubmit} />
      </Provider>
    );
  });

  it("should match the snapshot", () => {
    console.log(wrapper.debug());

    expect(wrapper).toMatchSnapshot();
  });

  it("State page is correct", () => {
    expect(state.page).toBe(pages.Statistics);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Provider store={store}>
        <Statistics submit={mockSubmit} />
      </Provider>,
      div
    );
  });

  it("should show text", () => {
    const text = wrapper.find({ id: "quizNumber" });
    expect(text.text()).toBe("QUIZ 1");
  });

  it("should find one button for guest", () => {
    const numOfQuestions = wrapper.find({ id: "quizNumber" });
    expect(numOfQuestions).toHaveLength(2);
  });
});
