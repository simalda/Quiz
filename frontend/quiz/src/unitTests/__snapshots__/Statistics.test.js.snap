// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stattistics should match the snapshot 1`] = `ShallowWrapper {}`;
